src/include/catalog/catalog_set.h

```diff
@@ -43,6 +48,12 @@ class CatalogSet {
     static std::unique_ptr<CatalogSet> deserialize(common::Deserializer& deserializer);
 
 private:
+    bool containsEntryNoLock(transaction::Transaction* transaction, const std::string& name) const;
+    CatalogEntry* getEntryNoLock(transaction::Transaction* transaction, const std::string& name);
+    void createEntryNoLock(transaction::Transaction* transaction,
+        std::unique_ptr<CatalogEntry> entry);
+    void dropEntryNoLock(transaction::Transaction* transaction, const std::string& name);
+
     void validateExist(transaction::Transaction* transaction, const std::string& name) const;
```

Comment:
Can u double check and rename private functions not grabbing a lock to with the `NoLock` suffix? e.g., `validateExist`, `validateNotExist`, `emplace`, `erase`, `traverseVersionChainsForTransaction`, `getCommittedEntry`. I think that's easier to read since we have some functions starting to have `NoLock` suffix now.